### PR TITLE
Update coredns and  prometheus config

### PR DESCRIPTION
**Pull request by zjj2wry**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator

```

### CODE REVIEW

File: charts/seed-monitoring/charts/core/charts/prometheus/templates/prometheus.yaml [hunk 1]
```diff
@@ -81,8 +81,7 @@ spec:
         - --config.file=/etc/prometheus/config/prometheus.yaml
         - --storage.tsdb.path=/var/prometheus/data
         - --storage.tsdb.no-lockfile
-        - --storage.tsdb.retention.time=30d
-        - --storage.tsdb.retention.size=15GB
+        - --storage.tsdb.retention.time=2h
```
Comment (tennix) [inline charts/seed-monitoring/charts/core/charts/prometheus/templates/prometheus.yaml:84]: Does this mean we can only tolerate 2 hours downtime for Prometheus?

File: charts/seed-monitoring/charts/core/charts/prometheus/rules/kube-prometheus-rules.yaml [hunk 2]
```diff
@@ -892,8 +892,8 @@ groups:
         }}) is restarting {{ printf "%.2f" $value }} times / 5 minutes.
```
Comment (tennix) [inline charts/seed-monitoring/charts/core/charts/prometheus/rules/kube-prometheus-rules.yaml:892]: The message need to be updated too.

File: charts/seed-monitoring/charts/core/charts/prometheus/templates/prometheus.yaml (hunk 1 above)
Comment (july2993) [reply charts/seed-monitoring/charts/core/charts/prometheus/templates/prometheus.yaml:84]: It means Prometheus will only retain data in the near 2h, so will not be able the query the data before 2h ago from now, seems not reasonable

Comment (tennix) [reply charts/seed-monitoring/charts/core/charts/prometheus/templates/prometheus.yaml:84]: The metrics will be collected by Thanos, so in normal case all the data will be searchable by Thanos. But I'm not sure if the Prometheus will truncate 2 hours data if it's down for more than 2 hours.

File: charts/shoot-core/components/charts/coredns/values.yaml [hunk 3]
```diff
@@ -44,7 +44,6 @@ configmap:
       parameters: cluster.local in-addr.arpa ip6.arpa
       configBlock: |-
         pods insecure
-        upstream
```
Comment (zjj2wry) [inline charts/shoot-core/components/charts/coredns/values.yaml:47]: https://github.com/aws/containers-roadmap/issues/1115

File: charts/seed-monitoring/charts/core/charts/prometheus/templates/prometheus.yaml (hunk 1 above)
Comment (zjj2wry) [reply charts/seed-monitoring/charts/core/charts/prometheus/templates/prometheus.yaml:84]: > It means Prometheus will only retain data in the near 2h, so will not be able the query the data before 2h ago from now, seems not reasonable

Prometheus data will be push to thanos-receive, prometheus only needs to maintain tsdb, thanos-receive will reject data when exceeding two hours.

> But I'm not sure if the Prometheus will truncate 2 hours data if it's down for more than 2 hours.

yes, but if thanos restart after 2hours, thanos will also reject data exceeding two hours.

**Review by tennix — APPROVED** [review]

**Review by aylei — APPROVED** [review]
lgtm

**Pull request merged**